.github/workflows/pipeline.yml

```diff
@@ -142,7 +154,32 @@ jobs:
       - name: Checkout code
         uses: actions/checkout@v4
 
-      - name: Build image from Dockerfile
+      - name: Restore Docker image cache
+        uses: actions/cache@v4
+        with:
+          path: |
+            /tmp/.docker-cache
+          key: ${{ runner.os }}-docker-${{ github.sha }}
+          restore-keys: |
+            ${{ runner.os }}-docker-
+
+      - name: Load cached Docker image
+        id: load-docker-image
+        run: |
+          IMAGE_PATH="/tmp/.docker-cache/${{ env.IMAGE_NAME }}_${{ github.sha }}.tar"
+
+          if docker load -i $IMAGE_PATH; then
+            echo "cache-hit=true" >> $GITHUB_ENV
+            echo "::set-output name=cache-hit::true"
```

Comment:
If this is a common way for naming, pls. ignore. I would have assume sth like "docker-image-load-successful" because the cache-hit/miss occurs in the step before and I got a little confused at first.

Reply:
The official `actions/cache@v4` action uses `cache-hit/miss` naming, so I reused this standard.

Reply:
>Happy to change to docker-image-load-successful if you'd like?

👍